dnf/conf/config.py

```diff
@@ -0,0 +1,929 @@
+        self.exists = exists
+        self.abspath = abspath
+        super(PathOption, self).__init__(default, parent, runtimeonly)
+
+    def _parse(self, val):
+        """Validate path."""
+        if val.startswith('file://'):
+            val = val[7:]
+        if self.abspath and val[0] != '/':
+            raise ValueError("Given path '%s' is not absolute." % val)
+        if self.exists and not os.path.exists(val):
+            raise ValueError("Given path '%s' does not exist." % val)
+        return val
+
+
+class IntOption(Option):
+    """An option representing an integer value."""
+
+    def __init__(self, default=None, parent=None, runtimeonly=False,
+                 range_min=None, range_max=None):
+        self._range_min = range_min
+        self._range_max = range_max
+        super(IntOption, self).__init__(default, parent, runtimeonly)
+
+    def _parse(self, s):
+        """Parse a string containing an integer."""
+        try:
+            n = int(s)
+        except (ValueError, TypeError):
+            raise ValueError('invalid integer value')
+
+        if self._range_max is not None and n > self._range_max:
+            raise ValueError('Given value [%d] should be less than '
+                             'allowed value [%d].' % (n, self._range_max))
+        if self._range_min is not None and n < self._range_min:
+            raise ValueError('Given value [%d] should be greater than '
+                             'allowed value [%d].' % (n, self._range_min))
+        return n
+
+
+class PositiveIntOption(IntOption):
+    """An option representing a positive integer value, where 0 can
+    have a special representation.
+    """
+    def __init__(self, default=None, parent=None, runtimeonly=False,
+                 range_min=0, range_max=None, names_of_0=None):
+        self._names0 = [] if names_of_0 is None else names_of_0
+        super(PositiveIntOption, self).__init__(default, parent, runtimeonly,
+                                                range_min, range_max)
+
+    def _parse(self, s):
+        """Parse a string containing a positive integer, where 0 can
+           have a special representation."""
+        if s in self._names0:
+            return 0
+        return super(PositiveIntOption, self)._parse(s)
+
+
+class SecondsOption(Option):
+    """An option representing an integer value of seconds, or a human
+    readable variation specifying days, hours, minutes or seconds
+    until something happens. Works like :class:`BytesOption`.  Note
+    that due to historical president -1 means "never", so this accepts
+    that and allows the word never, too.
+
+    Valid inputs: 100, 1.5m, 90s, 1.2d, 1d, 0xF, 0.1, -1, never.
+    Invalid inputs: -10, -0.1, 45.6Z, 1d6h, 1day, 1y.
+
+    Return value will always be an integer
+    """
+    MULTS = {'d': 60 * 60 * 24, 'h' : 60 * 60, 'm' : 60, 's': 1}
+
+    def _parse(self, s):
+        """Parse a string containing an integer value of seconds, or a human
```

Comment:
The description is quite same as under class. Probably same information is not necessary to repeat. Please can you change it.